File: app/src/main/java/com/BlizzardArmory/overwatch/Miscellaneous.java
package com.BlizzardArmory.overwatch;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.util.HashMap;

/**
 * The type Miscellaneous.
 */
public class Miscellaneous {

    @SerializedName("teleporterPadsDestroyed")
    @Expose
    private double teleporterPadsDestroyed;
    @SerializedName("turretsDestroyed")
    @Expose
    private double turretsDestroyed;

    /**
     * Gets teleporter pads destroyed.
     *
     * @return the teleporter pads destroyed
     */
    public double getTeleporterPadsDestroyed() {
        return teleporterPadsDestroyed;
    }

    /**
     * Sets teleporter pads destroyed.
     *
     * @param teleporterPadsDestroyed the teleporter pads destroyed
     */
    public void setTeleporterPadsDestroyed(double teleporterPadsDestroyed) {
        this.teleporterPadsDestroyed = teleporterPadsDestroyed;
    }

    /**
     * Gets turrets destroyed.
     *
     * @return the turrets destroyed
     */
    public double getTurretsDestroyed() {
        return turretsDestroyed;
    }

    /**
     * Sets turrets destroyed.
     *
     * @param turretsDestroyed the turrets destroyed
     */
    public void setTurretsDestroyed(double turretsDestroyed) {
        this.turretsDestroyed = turretsDestroyed;
    }

    /**
     * Gets misc.
     *
     * @return the misc
     */
    public HashMap<String, String> getMisc() {
        HashMap<String, String> miscList = new HashMap<>();
        NumberFormat formatter = new DecimalFormat("#0");

        miscList.put("TELEPORTER PADS DESTROYED", formatter.format(teleporterPadsDestroyed));
        miscList.put("TURRETS DESTROYED", formatter.format(turretsDestroyed));

        return miscList;
    }
}

File: app/src/main/java/com/BlizzardArmory/model/overwatch/Miscellaneous.kt
package com.BlizzardArmory.model.overwatch

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.*

/**
 * The type Miscellaneous.
 */
data class Miscellaneous(

        @SerializedName("teleporterPadsDestroyed")
        @Expose
        var teleporterPadsDestroyed: Double,

        @SerializedName("turretsDestroyed")
        @Expose
        var turretsDestroyed: Double
) {
    val misc: HashMap<String, String>
        get() {
            val miscList = HashMap<String, String>()
            val formatter: NumberFormat = DecimalFormat("#0")
            miscList["TELEPORTER PADS DESTROYED"] = formatter.format(teleporterPadsDestroyed)
            miscList["TURRETS DESTROYED"] = formatter.format(turretsDestroyed)
            return miscList
        }
}